src/ansible_compat/_compat.py

```diff
@@ -0,0 +1,31 @@
+"""Internal glue."""
+import logging
+import sys
+from typing import Callable, TypeVar
+
+try:
+    import tenacity
+except ImportError:
+    pass
+
+WrappedFn = TypeVar("WrappedFn", bound=Callable)
+_logger = logging.getLogger(__name__)
+
+
+def noop_dec(f: WrappedFn) -> WrappedFn:
+    """Decorate function with a no-op."""
+    return f
+
+
+if "tenacity" in sys.modules:
+    retry = tenacity.retry(  # Retry up to 3 times as galaxy server can return errors
+        reraise=True,
+        wait=tenacity.wait_fixed(30),  # type: ignore
+        stop=tenacity.stop_after_attempt(3),  # type: ignore
+        before_sleep=tenacity.after_log(_logger, logging.WARNING),  # type: ignore
+    )
+else:
+    retry = noop_dec
+
+
+__all__ = ["retry"]
```

Comment:
This must be immutable. Normally, pylint warns about this but that check seems to be disabled for some reason.
```suggestion
__all__ = "retry",
```